API/hermes_abi/hermes_abi.h

```diff
@@ -0,0 +1,400 @@
+      HermesABIContext *,
+      const uint8_t *,
+      size_t);
+  HermesABIPropNameIDOrError (
+      *create_prop_name_id_from_string)(HermesABIContext *, HermesABIString);
+  HermesABIPropNameIDOrError (
+      *create_prop_name_id_from_symbol)(HermesABIContext *, HermesABISymbol);
+  bool (*compare_prop_name_id)(
+      HermesABIContext *,
+      HermesABIPropNameID,
+      HermesABIPropNameID);
+
+  bool (*object_is_array)(HermesABIContext *, HermesABIObject);
+  bool (*object_is_array_buffer)(HermesABIContext *, HermesABIObject);
+  bool (*object_is_function)(HermesABIContext *, HermesABIObject);
+  bool (*object_is_host_object)(HermesABIContext *, HermesABIObject);
+  bool (*function_is_host_function)(HermesABIContext *, HermesABIFunction);
+
+  HermesABIValueOrError (*call)(
+      HermesABIContext *,
+      HermesABIFunction,
+      const HermesABIValue *,
+      const HermesABIValue *,
+      size_t);
+  HermesABIValueOrError (*call_as_constructor)(
+      HermesABIContext *,
+      HermesABIFunction,
+      const HermesABIValue *,
+      size_t);
+
+  HermesABIFunctionOrError (*create_function_from_host_function)(
+      HermesABIContext *,
+      HermesABIPropNameID,
+      unsigned int,
+      HermesABIHostFunction *);
+  HermesABIHostFunction *(
+      *get_host_function)(HermesABIContext *, HermesABIFunction);
+
+  HermesABIObjectOrError (*create_object_from_host_object)(
+      HermesABIContext *,
+      HermesABIHostObject *);
+  HermesABIHostObject *(*get_host_object)(HermesABIContext *, HermesABIObject);
+
+  bool (*has_native_state)(HermesABIContext *, HermesABIObject);
+  HermesABINativeState *(
+      *get_native_state)(HermesABIContext *, HermesABIObject);
+  HermesABIVoidOrError (*set_native_state)(
+      HermesABIContext *,
+      HermesABIObject,
+      HermesABINativeState *);
+
+  HermesABIWeakObjectOrError (
+      *create_weak_object)(HermesABIContext *, HermesABIObject);
+  HermesABIValue (*lock_weak_object)(HermesABIContext *, HermesABIWeakObject);
+
+  void (*get_utf8_from_string)(
+      HermesABIContext *,
+      HermesABIString,
+      HermesABIByteBuffer *);
+  void (*get_utf8_from_prop_name_id)(
+      HermesABIContext *,
+      HermesABIPropNameID,
+      HermesABIByteBuffer *);
+  void (*get_utf8_from_symbol)(
+      HermesABIContext *,
+      HermesABISymbol,
+      HermesABIByteBuffer *);
+
+  HermesABIBoolOrError (
+      *instance_of)(HermesABIContext *, HermesABIObject, HermesABIFunction);
+
+  bool (*strict_equals_symbol)(
+      HermesABIContext *,
+      HermesABISymbol,
+      HermesABISymbol);
+  bool (*strict_equals_bigint)(
+      HermesABIContext *,
+      HermesABIBigInt,
+      HermesABIBigInt);
+  bool (*strict_equals_string)(
+      HermesABIContext *,
+      HermesABIString,
+      HermesABIString);
+  bool (*strict_equals_object)(
+      HermesABIContext *,
+      HermesABIObject,
+      HermesABIObject);
+};
+
+#ifdef _MSC_VER
+__declspec(dllexport)
```

Comment:
We need a flag here to avoid exporting if we consume this header from other code.

Reply:
Why?

I'm asking because the existing Hermes and JSI headers always include the visibility specifier.

Reply:
It maybe not a big deal for Clang, but for MSVC we must be very specific.
The DLL that exports a function uses `__declspec(dllexport)`.
Then, code that imports the function must use `__declspec(dllimport)` or maybe nothing.
Using `__declspec(dllexport)` in user code is an error and linker may complain about it.